webpack.config.js

```diff
@@ -27,18 +27,19 @@ const gutenbergPackages = Object.keys( dependencies )
 	.filter( ( packageName ) => packageName.startsWith( WORDPRESS_NAMESPACE ) )
 	.map( ( packageName ) => packageName.replace( WORDPRESS_NAMESPACE, '' ) );
 
-config.entry = gutenbergPackages.reduce( ( memo, packageName ) => {
-	const name = camelCaseDash( packageName );
-	memo[ name ] = `./packages/${ packageName }`;
-	return memo;
-}, {} );
-
-config.output = {
-	filename: './build/[basename]/index.js',
-	path: __dirname,
-	library: [ 'wp', '[name]' ],
-	libraryTarget: 'this',
-};
+const config = Object.assign( {}, defaultConfig, {
```

Comment:
Here's a proposal using the spread syntax that also avoids mutating the `plugins` array in defaultConfig:

```js
const config = { ...defaultConfig, ...{
    entry: // new entry here
} };

// Make sure we don't mutate the defaultConfig.plugins array
config.plugins = [ ...defaultConfig.plugins ];
config.plugins.push( /* new plugin */ );
```

I'd include the comment as this is easy to overlook.

Reply:
it's probably use something which performs deep clone instead: https://lodash.com/docs/4.17.11#cloneDeep.

Reply:
It looks like the object spread is supported since node 8.6.0 so we can start using in CLI scripts:
https://node.green/#ES2018-features-object-rest-spread-properties

Reply:
I have updated code to use object spread for config and array spread for plugins which seems to work with Node 10 🎉